kolibri/logger/api.py

```diff
@@ -1,35 +1,89 @@
 from kolibri.auth.api import KolibriAuthPermissions, KolibriAuthPermissionsFilter
+from kolibri.auth.models import Classroom, LearnerGroup
 from kolibri.content.api import OptionalPageNumberPagination
-from rest_framework import viewsets
+from rest_framework import filters, viewsets
 
 from .models import ContentInteractionLog, ContentRatingLog, ContentSummaryLog, UserSessionLog
 from .serializers import ContentInteractionLogSerializer, ContentRatingLogSerializer, ContentSummaryLogSerializer, UserSessionLogSerializer
 
 
+class BaseLogFilter(filters.FilterSet):
+    facility = filters.django_filters.MethodFilter()
+    classroom = filters.django_filters.MethodFilter()
+    learner_group = filters.django_filters.MethodFilter()
+
+    # Only device owner (superuser) can filter by facilities
+    def filter_facility(self, queryset, value):
+        return queryset.filter(user__facility_id=value)
+
+    def filter_classroom(self, queryset, value):
+        classroom_members_pks = Classroom.objects.get(pk=value).get_members().values_list("id", flat=True)
+        return queryset.filter(user__pk__in=list(classroom_members_pks))
```

Comment:
Changing these to the following (could be the same, actually, for classroom and learner_group) makes it 12 times faster (I just tested):

```
        return HierarchyRelationsFilter(queryset).filter_by_hierarchy(
            ancestor_collection=value,
            target_user=F("user"),
        )
```

You'll need to add these at the top:
```
from django.db.models.query import F
from kolibri.auth.filters import HierarchyRelationsFilter
```